langchain/document_loaders/arxiv.py

```diff
@@ -14,18 +14,19 @@ class ArxivLoader(BaseLoader):
 
     def __init__(
         self,
-        query: str,
+        query: Optional[str] = None,
         load_max_docs: Optional[int] = 100,
         load_all_available_meta: Optional[bool] = False,
     ):
         self.query = query
         self.load_max_docs = load_max_docs
         self.load_all_available_meta = load_all_available_meta
 
-    def load(self) -> List[Document]:
+    def load(self, query: Optional[str] = None) -> List[Document]:
         arxiv_client = ArxivAPIWrapper(
             load_max_docs=self.load_max_docs,
             load_all_available_meta=self.load_all_available_meta,
         )
-        docs = arxiv_client.load(self.query)
+        query = query or self.query
```

Comment:
Yes, I agree. Still, the Class initiation parameters naturally have lower precedence in regard to the calling parameters.
I'd rather completely move `query` from `__init__`. But it will be a breaking change.